vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt

```diff
@@ -196,6 +199,24 @@ class RoomDetailViewModel @AssistedInject constructor(
         if (OutboundSessionKeySharingStrategy.WhenEnteringRoom == BuildConfig.outboundSessionKeySharingStrategy && room.isEncrypted()) {
             prepareForEncryption()
         }
+
+        if (initialState.switchToParentSpace) {
+            // We are coming from a notification, try to switch to the most relevant space
+            // so that when hitting back the room will appear in the list
+            appStateHandler.getCurrentRoomGroupingMethod()?.space().let { currentSpace ->
+                val currentRoomSummary = room.roomSummary() ?: return@let
+                // nothing we are good
+                if (currentSpace == null || !currentRoomSummary.flattenParentIds.contains(currentSpace.roomId)) {
+                    // take first one or switch to home
+                    appStateHandler.setCurrentSpace(
+                            currentRoomSummary
+                                    .flattenParentIds.firstOrNull { it.isNotBlank() },
```

Comment:
Also why taking the first space here?

Reply:
It's not that trivial to know which one to take. We could check and take the root parent (in case of part of a subspace)?